### PR TITLE
feat: add datetime package

### PR DESCRIPTION
**Pull request by tothandras**
## Summary by CodeRabbit

* **New Features**
  * Introduced advanced date-time and ISO 8601 duration handling, including parsing, formatting, arithmetic, and validation with timezone support.
  * Added support for multiple date-time formats (RFC3339, ISO8601, RFC9557) and precise duration arithmetic.
  * Added JSONB annotations field to usage resets with full create, update, and query support.
  * Added PostgreSQL migration for usage period duration calculations with normalized interval handling and ULID generation.
  * Enhanced error handling for datetime and duration parsing and operations.

* **Bug Fixes**
  * Improved consistency and accuracy in duration parsing and representation across the application.

* **Tests**
  * Added extensive unit tests for date-time and duration parsing, formatting, arithmetic, and edge cases.
  * Added comprehensive migration tests validating usage period interval backfills and normalization.

* **Chores**
  * Updated tests and internal usage to use new duration parsing and construction methods.
  * Removed deprecated or redundant test utilities.
  * Refactored recurrence intervals to use ISO duration types.

### CODE REVIEW

File: pkg/datetime/datetime_test.go [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+			// Format the parsed time back to a string
+			formatted := parsed.Format(tt.layout)
+
+			// Verify round-trip: the input time should equal the formatted time
+			assert.Equal(t, tt.input, formatted, "round-trip formatting should preserve the original input")
+		})
+	}
+}
+
+func TestDateTime_Add(t *testing.T) {
```
Comment (GAlexIHU) [inline pkg/datetime/datetime_test.go]: nit: cursor's default test name format is different to how we do it

File: pkg/datetime/duration.go [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+				return false, nil
+			}
+		}
+	}
+
+	return true, nil
+}
+
+// convertPeriodToSeconds converts a period to total seconds using decimal precision
+func convertPeriodToSeconds(p period.Period, daysInMonth int, hoursInDays int) (decimal.Decimal, error) {
```
Comment (GAlexIHU) [inline pkg/datetime/duration.go:169]: lets try to stick with the libs we already use and use alpacadecimal for this (as we already do)

**Comment by GAlexIHU** [comment]
Merge this first (into this) https://github.com/openmeterio/openmeter/pull/3088

**Review by turip — APPROVED** [review]
